src/test/java/com/sonymobile/jenkins/plugins/lenientshutdown/GlobalLenientShutdownTest.java

```diff
@@ -144,6 +150,22 @@ public void testBlocksBuildWhenShutdownEnabled() throws Exception {
     }
 
     /**
+     * Tests that pipleine job is blocked after shudown mode is initiated.
+     * @throws Exception if something goes wrong
+     */
+    @Test
+    public void testPipelineIsBlockedWhenShutdownEnabled() throws Exception {
+        WorkflowJob project = jenkinsRule.jenkins.createProject(WorkflowJob.class, "p");
+        toggleLenientShutdown();
+
+        project.scheduleBuild2(0);
+        Item queueItem = waitForBlockedItem(project, TIMEOUT_SECONDS);
+
+        assertThat(queueItem.isBlocked(), is(true));
```

Comment:
I do not think it should work like this. `WorkflowRun`s should not be blocked since they can survive a restart.